manageiq-operator/pkg/helpers/miq-components/postgresql.go

```diff
@@ -173,64 +182,68 @@ func NewPostgresqlDeployment(cr *miqv1alpha1.ManageIQ) (*appsv1.Deployment, erro
 			corev1.VolumeMount{Name: "miq-pg-configs", MountPath: "/opt/app-root/src/postgresql-cfg/"},
 		},
 	}
+}
 
-	err := addResourceReqs(cr.Spec.PostgresqlMemoryLimit, cr.Spec.PostgresqlMemoryRequest, cr.Spec.PostgresqlCpuLimit, cr.Spec.PostgresqlCpuRequest, &container)
-	if err != nil {
-		return nil, err
+func postgresqlPodSpec(cr *miqv1alpha1.ManageIQ) corev1.PodSpec {
+
+	return corev1.PodSpec{
+		Volumes: []corev1.Volume{
+			corev1.Volume{
+				Name: "miq-pgdb-volume",
+				VolumeSource: corev1.VolumeSource{
+					PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+						ClaimName: "postgresql",
+					},
+				},
+			},
+			corev1.Volume{
+				Name: "miq-pg-configs",
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{Name: "postgresql-configs"},
+					},
+				},
+			},
+		},
 	}
 
-	deploymentLabels := map[string]string{
-		"app": cr.Spec.AppName,
+}
+
+func NewPostgresqlDeployment(cr *miqv1alpha1.ManageIQ) (*appsv1.Deployment, controllerutil.MutateFn, error) {
+
+	container := postgresqlContainer(cr)
```

Comment:
Why did you split this out from the pod spec function? If you keep this in the same function then I think this will be a lot simpler.